### PR TITLE
Slider: refactor using hooks

### PR DESCRIPTION
**Pull request by CloudPower97**
# Description

Part of #928 

# How Has This Been Tested?

Removed unnecessary `spec`

# Checklist:

- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings
- [x] I have not generated a new package version. (the maintainers will handle that)

### CODE REVIEW

**Comment by alextrastero** [comment]
Could you merge master into this branch?
`git merge master --no-ff`

File: src/Slider.js [hunk 1]
```diff
@@ -1,80 +1,30 @@
...
+  useEffect(() => {
+    const instance = M.Slider.init(_slider.current, options);
 
-  render() {
-    const { fullscreen, children, className } = this.props;
-    const classes = {
-      fullscreen,
-      slider: true
+    return () => {
+      instance.current && instance.current.destroy();
```
Comment (alextrastero) [inline src/Slider.js]: this is just `instance && instance.destroy()`

Comment (CloudPower97) [reply src/Slider.js]: Whoops, my bad

**Comment by CloudPower97** [comment]
Hey @alextrastero, I'd like to hear your thoughts.
Currently, we have this code:
```js
  useEffect(() => {
    const instance = M.Slider.init(_slider.current, options);

    if (activeIndex) {
      if (typeof indicators === 'undefined' || options.indicators) {
        instance['$indicators'][activeIndex].className =
          'indicator-item active';
      }
    }

    return () => {
      if (instance) {
        setActiveIndex(instance.activeIndex);
        instance.destroy();
      }
    };
  }, [_slider, options, fullscreen, activeIndex]);
```
an alternative approach would be this:

```js
const sliderInstance = useRef(null);

  useEffect(() => {
    sliderInstance.current = M.Slider.init(_slider.current, options);

    return () => {
      if (sliderInstance.current) {
        setActiveIndex(sliderInstance.current.activeIndex);
        sliderInstance.current.destroy();
      }
    };
  }, [_slider, options, fullscreen]);

  useEffect(() => {
    if (activeIndex) {
      if (typeof indicators === 'undefined' || options.indicators) {
        sliderInstance.current['$indicators'][activeIndex].className =
          'indicator-item active';
      }
    }
  }, [activeIndex, options.indicators, fullscreen]);
```

I think the first one is cleaner, but actually it seems that `React` encourages to separate as much logic as possible in small reusable `hooks` instead of a bigger one.

So, I'm not sure what to do, and I'd really appreciate a your suggestion.

**Comment by alextrastero** [comment]
Yes, agree with separating in smaller useEffects :)

**Comment by CloudPower97** [comment]
ping @alextrastero :smile:

**Pull request merged**